src/client.rs

```diff
@@ -157,16 +168,51 @@ impl DeezerClient {
         Ok(res.data)
     }
 
-    async fn get<T: DeserializeOwned>(&self, url: &str) -> Result<T> {
-        let res = self
+    pub(crate) async fn get_subresource<T>(&self, id: u64, limit: Option<u32>,
+                                           offset: Option<u32>) -> Result<Vec<T>>
+        where
+            T: DeezerObject
+    {
+        let url = T::get_api_url(id);
+        let url = format!("{}/{}", BASE_URL, url);
+
+        let mut params: HashMap<String, String> = HashMap::new();
+        if let Some(_limit) = limit {
+            params.insert("limit".to_owned(), _limit.to_string());
+        }
+        if let Some(_offset) = offset {
+            params.insert("offset".to_owned(), _offset.to_string());
+        }
+
+        let res: DeezerArray<T> = self.get_with_params(&url, &params).await?;
+
+        Ok(res.data)
+    }
+
+    async fn get_with_optional_params<T: DeserializeOwned>(&self, url: &str, query_params: Option<&HashMap<String, String>>) -> Result<T> {
+        let mut request_builder = self
             .client
-            .get(url)
-            .send()
-            .await?
-            .error_for_status()?
-            .json()
-            .await?;
+            .get(url);
+        if query_params.is_some() {
+            request_builder = request_builder.query(query_params.unwrap());
+        }
```

Comment:
Here you can use pattern matching instead of unwraping like so:
```suggestion
        if let Some(params) = query_params {
            request_builder = request_builder.query(params);
        }
```

Reply:
changed